src/ui/SortDropdown/SortDropdown.ts

```diff
@@ -0,0 +1,159 @@
+    this.bind.onQueryState(MODEL_EVENTS.CHANGE_ONE, QUERY_STATE_ATTRIBUTES.SORT, (args: IAttributeChangedEventArg) =>
+      this.handleQueryStateChanged(args)
+    );
+    this.bind.onRootElement(QueryEvents.querySuccess, (args: IQuerySuccessEventArgs) => this.handleQuerySuccess(args));
+    this.bind.onRootElement(QueryEvents.queryError, (args: IQueryErrorEventArgs) => this.handleQueryError(args));
+  }
+
+  private handleAfterInitialization() {
+    this.buildDropdown();
+  }
+
+  private buildDropdown() {
+    this.sortItems = this.getSortItems();
+    this.dropdown = new Dropdown(
+      () => this.handleChange(),
+      this.getValuesForDropdown(),
+      value => this.getCaptionForValue(value),
+      l('SortBy')
+    );
+    this.element.appendChild(this.dropdown.getElement());
+    this.update();
+  }
+
+  private getSortItems() {
+    const sortItems: ISortDropdownItem[] = [];
+    const sortElements = $$(this.element).findAll('.CoveoSort');
+
+    each(sortElements, sortEl => {
+      const sortInstance = <Sort>Component.get(sortEl, Sort);
+
+      if (sortInstance.options.sortCriteria.length > 1) {
+        this.logger.warn(
+          `Each Sort component inside a SortDropdown should have only one sort criteria. Not using ${
+            sortInstance.options.caption
+          } in the SortDropdown.`
+        );
+      } else {
+        sortItems.push({
+          caption: sortInstance.options.caption,
+          criteria: sortInstance.options.sortCriteria[0]
+        });
+      }
+    });
+
+    return sortItems;
+  }
+
+  private getValuesForDropdown(): string[] {
+    return this.sortItems.map(item => item.criteria.toString());
+  }
+
+  private handleQueryStateChanged(data: IAttributeChangedEventArg) {
+    this.update();
+  }
+
+  private update() {
+    if (!this.dropdown) {
+      return;
+    }
+
+    const sortCriteria = <string>this.queryStateModel.get(QueryStateModel.attributesEnum.sort);
+    const itemIndex = this.getSortItemIndex(sortCriteria);
+    itemIndex > -1 && this.dropdown.select(itemIndex, false);
+
+    $$(this.dropdown.getElement()).toggleClass('coveo-selected', itemIndex > -1);
+  }
+
+  private handleQuerySuccess(data: IQuerySuccessEventArgs) {
+    data.results.results.length == 0 ? this.hideElement() : this.showElement();
+  }
+
+  private handleQueryError(data: IQueryErrorEventArgs) {
+    this.hideElement();
+  }
+
+  private handleChange() {
+    const selectedValue = this.dropdown.getValue();
+    $$(this.element)
+      .find(`.CoveoSort[data-sort-criteria="${selectedValue}"]`)
+      .click();
+  }
+
+  private getCaptionForValue(value: string) {
+    const index = this.getSortItemIndex(value);
+    return index > -1 ? this.sortItems[index].caption : '';
+  }
+
+  private getSortItemIndex(itemValue: string) {
+    return findIndex(this.sortItems, sortItem => sortItem.criteria.toString() === itemValue);
+  }
+
+  private hideElement() {
+    $$(this.element).addClass('coveo-hidden');
```

Comment:
You're right. Still not sure why the `coveo-hidden` way is used that much in the codebase.

Reply:
Mmm it seems `coveo-hidden` classes are used a lot because that's how visibility is unit tested. Not using a class would require something like `$.is(':visible')` but I think is not availble in our `$$` implementation. So I'll keep the css class approach as one of the unit tests should check the component is shown/hidden depending on querySuccess/queryError/number of results.